### PR TITLE
Fix case collision between FLAKE and Flake.

### PR DESCRIPTION
**Pull request by claunia**
On Windows there is no difference but the project cannot be compiled on Linux.

### CODE REVIEW

**Comment by c72578** [comment]
Thanks for your PR.
Could you please describe in detail, which steps to carry out, to reproduce the case collision between FLAKE and Flake.
OS and programs used ...
In order to test this PR.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public class FlaCudaWriter : IAudioDest
 		TimeSpan _userProcessorTime;
 
 		// header bytes
-		// allocated by flake_encode_init and freed by flake_encode_close
+		// allocated by Flake_encode_init and freed by Flake_encode_close
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:108]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 2]
```diff
@@ -156,14 +156,14 @@ public FlaCudaWriter(string path, Stream IO, AudioPCMConfig pcm)
 			sample_rate = pcm.SampleRate;
 			bits_per_sample = (uint) pcm.BitsPerSample;
 
-			// flake_validate_params
+			// Flake_validate_params
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:159]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.
Furthermore `flake_validate_params` has been commented out throughout the project and is not relevant to this issue.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 3]
```diff
@@ -156,14 +156,14 @@ public FlaCudaWriter(string path, Stream IO, AudioPCMConfig pcm)
...
-			// flake_validate_params
+			// Flake_validate_params
 
 			_path = path;
 			_IO = IO;
 
 			windowBuffer = new float[FlaCudaWriter.MAX_BLOCKSIZE * lpc.MAX_LPC_WINDOWS];
 
-			eparams.flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
+			eparams.Flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:166]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 4]
```diff
@@ -206,7 +206,7 @@ public int CompressionLevel
 				if (value < 0 || value > 11)
 					throw new Exception("unsupported compression level");
 				_compressionLevel = value;
-				eparams.flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
+				eparams.Flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:209]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 5]
```diff
@@ -223,7 +223,7 @@ public object Settings
 				if (value as FlaCudaWriterSettings == null)
 					throw new Exception("Unsupported options " + value);
 				_settings = value as FlaCudaWriterSettings;
-				eparams.flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
+				eparams.Flake_set_defaults(_compressionLevel, !_settings.GPUOnly);
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:226]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 6]
```diff
@@ -1574,7 +1574,7 @@ public unsafe void InitTasks()
 				//cuda.LoadModule(System.IO.Path.Combine(Environment.CurrentDirectory, "flacuda.cubin"));
 				if (_IO == null)
 					_IO = new FileStream(_path, FileMode.Create, FileAccess.Write, FileShare.Read);
-				int header_size = flake_encode_init();
+				int header_size = Flake_encode_init();
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1577]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 7]
```diff
@@ -1880,11 +1880,11 @@ int write_headers()
 			return header_size;
 		}
 
-		int flake_encode_init()
+		int Flake_encode_init()
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1883]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 8]
```diff
@@ -1880,11 +1880,11 @@ int write_headers()
 			return header_size;
 		}
 
-		int flake_encode_init()
+		int Flake_encode_init()
 		{
 			int i, header_len;
 
-			//if(flake_validate_params(s) < 0)
+			//if(Flake_validate_params(s) < 0)
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1887]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.
Furthermore `flake_validate_params` has been commented out throughout the project and is not relevant to this issue.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 9]
```diff
@@ -1963,7 +1963,7 @@ int flake_encode_init()
 	struct FlakeEncodeParams
 	{
 		// compression quality
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1966]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 10]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
 		public int compression;
 
 		// stereo decorrelation method
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1975]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 11]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 0 to 2
 		// 0 = independent L+R channels
 		// 1 = mid-side encoding
 		public bool do_midside;
 
 		// block size in samples
-		// set by the user prior to calling flake_encode_init
+		// set by the user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1983]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 12]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to 0, a block size is chosen based on block_time_ms
 		// can also be changed by user before encoding a frame
 		public int block_size;
 
 		// block time in milliseconds
-		// set by the user prior to calling flake_encode_init
+		// set by the user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1989]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 13]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// used to calculate block_size based on sample rate
 		// can also be changed by user before encoding a frame
 		public int block_time_ms;
 
 		// padding size in bytes
-		// set by the user prior to calling flake_encode_init
+		// set by the user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:1995]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 14]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, defaults to 4096
 		public long padding_size;
 
 		// minimum LPC order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2000]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 15]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 1 to 32
 		public int min_prediction_order;
 
 		// maximum LPC order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2006]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 16]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 1 to 32 
 		public int max_prediction_order;
 
 		public int orders_per_window;
 
 		// minimum fixed prediction order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2014]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 17]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 0 to 4
 		public int min_fixed_order;
 
 		// maximum fixed prediction order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2020]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 18]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 0 to 4 
 		public int max_fixed_order;
 
 		// minimum partition order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2026]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 19]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 0 to 8
 		public int min_partition_order;
 
 		// maximum partition order
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2032]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 20]
```diff
@@ -1972,70 +1972,70 @@ struct FlakeEncodeParams
...
 		// if set to less than 0, it is chosen based on compression.
 		// valid values are 0 to 8
 		public int max_partition_order;
 
 		// whether to use variable block sizes
-		// set by user prior to calling flake_encode_init
+		// set by user prior to calling Flake_encode_init
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2038]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools.Codecs.FlaCuda/FlaCudaWriter.cs [hunk 21]
```diff
@@ -2055,7 +2055,7 @@ struct FlakeEncodeParams
 
 		public bool do_seektable;
 
-		public int flake_set_defaults(int lvl, bool encode_on_cpu)
+		public int Flake_set_defaults(int lvl, bool encode_on_cpu)
```
Comment (c72578) [inline CUETools.Codecs.FlaCuda/FlaCudaWriter.cs:2058]: Please undo. The lower case letter at the beginning of the method name should stay, like in other methods.

File: CUETools/CUETools.TestCodecs/FlakeWriterTest.cs [hunk 22]
```diff
@@ -79,24 +79,24 @@ public void ConstructorTest()
             AudioBuffer buff = Codecs.WAV.AudioDecoder.ReadAllSamples(new Codecs.WAV.DecoderSettings(), "test.wav");
             AudioEncoder target;
 
-            target = new AudioEncoder(new EncoderSettings() { PCM = buff.PCM, EncoderMode = "7" }, "flakewriter0.flac");
+            target = new AudioEncoder(new EncoderSettings() { PCM = buff.PCM, EncoderMode = "7" }, "Flakewriter0.flac");
```
Comment (c72578) [inline CUETools/CUETools.TestCodecs/FlakeWriterTest.cs:82]: No need to change the file name of `flakewriter0.flac` in the `FlakeWriterTest`. It should stay lower case, in order to avoid unnecessary changes, which are not related to the issue.

File: CUETools/CUETools.TestCodecs/FlakeWriterTest.cs [hunk 23]
```diff
@@ -79,24 +79,24 @@ public void ConstructorTest()
...
             target.Settings.Padding = 1;
             target.DoSeekTable = false;
             //target.Vendor = "CUETools";
             //target.CreationTime = DateTime.Parse("15 Aug 1976");
             target.FinalSampleCount = buff.Length;
             target.Write(buff);
             target.Close();
-            CollectionAssert.AreEqual(File.ReadAllBytes("flake.flac"), File.ReadAllBytes("flakewriter0.flac"), "flakewriter0.flac doesn't match.");
+            CollectionAssert.AreEqual(File.ReadAllBytes("Flake.flac"), File.ReadAllBytes("Flakewriter0.flac"), "Flakewriter0.flac doesn't match.");
```
Comment (c72578) [inline CUETools/CUETools.TestCodecs/FlakeWriterTest.cs:90]: No need to change the file names of `flake.flac` and `flakewriter0.flac` in the `FlakeWriterTest`. They should stay lower case, in order to avoid unnecessary changes, which are not related to the issue.

File: CUETools/CUETools.TestCodecs/FlakeWriterTest.cs [hunk 24]
```diff
@@ -79,24 +79,24 @@ public void ConstructorTest()
...
 
-            target = new AudioEncoder(new EncoderSettings() { PCM = buff.PCM, EncoderMode = "7" }, "flakewriter1.flac");
+            target = new AudioEncoder(new EncoderSettings() { PCM = buff.PCM, EncoderMode = "7" }, "Flakewriter1.flac");
```
Comment (c72578) [inline CUETools/CUETools.TestCodecs/FlakeWriterTest.cs:92]: No need to change the file name of `flakewriter1.flac` in the `FlakeWriterTest`. It should stay lower case, in order to avoid unnecessary changes, which are not related to the issue.

File: CUETools/CUETools.TestCodecs/FlakeWriterTest.cs [hunk 25]
```diff
@@ -79,24 +79,24 @@ public void ConstructorTest()
...
             target.Settings.Padding = 1;
             target.DoSeekTable = false;
             //target.Vendor = "CUETools";
             //target.CreationTime = DateTime.Parse("15 Aug 1976");
             target.Write(buff);
             target.Close();
-            CollectionAssert.AreEqual(File.ReadAllBytes("flake.flac"), File.ReadAllBytes("flakewriter1.flac"), "flakewriter1.flac doesn't match.");
+            CollectionAssert.AreEqual(File.ReadAllBytes("Flake.flac"), File.ReadAllBytes("Flakewriter1.flac"), "Flakewriter1.flac doesn't match.");
```
Comment (c72578) [inline CUETools/CUETools.TestCodecs/FlakeWriterTest.cs:99]: No need to change the file names of `flake.flac` and `flakewriter1.flac` in the `FlakeWriterTest`. They should stay lower case, in order to avoid unnecessary changes, which are not related to the issue.

**Comment by claunia** [comment]
Hi,

It was not my intention to rename those methods I will fix that.

To test it you need to compile on Linux, or use HFSX or Case-sensitive APFS in macOS (that is known to cause some problems) or enable NTFS case sensitiveness (don't do it in C: or most things will just stop working).

**Comment by ipaucek4680** [comment]
@claunia 
Please try #56.

Note that when I attempt to compile CueTools on Linux, it looks like the compiler will complain about some (Windows-specific?) codes/libraries, such as the RunOnce code in CUERipper.

**Comment by c72578** [comment]
@claunia Are you happy to update this PR, so that it only contains the necessary changes, as commented in the review above?

**Comment by c72578** [comment]
@claunia, I have updated the PR now, so that it only contains the necessary changes.

**Comment by c72578** [comment]
@claunia, thanks for your contribution.

**Comment by claunia** [comment]
@c72578 Sorry I've been busy and couldn't attend to this. Thanks to you for the FLAC implementation!

**Comment by awsms** [comment]
@claunia were you actually able to compile it on linux?
